cclib/parser/molcasparser.py

```diff
@@ -441,6 +443,141 @@ def extract(self, inputfile, line):
+		#    2   -382.30702964 -0.00679742 0.043573  0.014908 nrc001   0.068195  nrc001     -382.30871333 RS-RFO  BFGS      0    
+		#    3   -382.30805348 -0.00102384 0.014883  0.005458 nrc010  -0.020973  nrc001     -382.30822089 RS-RFO  BFGS      0    
+		# ...
+		# ...
+		#   18   -382.30823419 -0.00000136 0.001032  0.000100 nrc053   0.012319  nrc053     -382.30823452 RS-RFO  BFGS      0    
+		#   19   -382.30823198  0.00000221 0.001051 -0.000092 nrc054   0.066565  nrc053     -382.30823822 RS-RFO  BFGS      0    
+		#   20   -382.30820252  0.00002946 0.001132 -0.000167 nrc021  -0.064003  nrc053     -382.30823244 RS-RFO  BFGS      0    
+		#
+		#         +----------------------------------+----------------------------------+
+		#         +    Cartesian Displacements       +    Gradient in internals         +
+		#         +  Value      Threshold Converged? +  Value      Threshold Converged? +
+		#   +-----+----------------------------------+----------------------------------+
+		#   + RMS + 5.7330E-02  1.2000E-03     No    + 1.6508E-04  3.0000E-04     Yes   +
+		#   +-----+----------------------------------+----------------------------------+
+		#   + Max + 1.2039E-01  1.8000E-03     No    + 1.6711E-04  4.5000E-04     Yes   +
+		#   +-----+----------------------------------+----------------------------------+
+        if 'Energy Statistics for Geometry Optimization' in line:
+            if not hasattr(self, 'geovalues'):
+                self.geovalues = []
+
+            self.skip_line(inputfile, 'stars')
+            while 'Iter      Energy       Change     Norm' not in line:
+                line = next(inputfile)
+            
+            # A variable keeping track of ongoing iteration.
+            iter_number = len(self.geovalues) + 1
+            # Iterate till blank line.
+            while line.split() != []:
+                for i in range(iter_number):
+                    line = next(inputfile)
+                self.geovalues.append([float(line.split()[2])])
+                line = next(inputfile)
+            # ALong with ENergy change, RMS and Max values of change in 
+            # Cartesian Diaplacement and Gradients are used as optimization
+            # criteria. 
+            while '+ RMS +' not in line:
+                line = next(inputfile)
+            
+            line_rms = line.split()
+            line = next(inputfile)
+            line_max = next(inputfile).split()
+            if not hasattr(self, 'geotargets'):
+                # The attribute geotargets is an array consisting of the following
+                # values: [Energy threshold, RMS Displacements threshold, RMS Gradient threshold, \
+                #          Max Displacements threshold, Max Gradient threshold].
+                self.geotargets = [1E-06, float(line_rms[4]), float(line_rms[8]), float(line_max[4]), float(line_max[4])]
```

Comment:
1. Is it really fixed at 1.0e-06? If it's printed somewhere, it should be parsed instead.
2. Are you sure it's 4, 8, 4, 4 and not 4, 8, 4, 8?

Reply:
For number 1. I couldn't find it printed anywhere. Will take another look at the file. What exactly are these energy values?

Reply:
See https://molcas.gitlab.io/doc/sphinx/users.guide/programs/slapaf.html for all the details on the geometry optimizer. Specifically, see `THRShld`.

The energy threshold they mention is printed under `Slapaf input parameters` as `Convergence criterion on energy change`, which would be theta here: ![codecogseqn](https://user-images.githubusercontent.com/727571/41168793-f96852b4-6b14-11e8-9db6-9520757e034a.gif)